conductr_cli/test/test_bytes_util.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2010 Jason Moiron and Contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+from unittest import TestCase
+from conductr_cli import bytes_util
+
+
+class TestBytesUtil(TestCase):
+    def test_naturalsize(self):
+        tests = (300, 3000, 3000000, 3000000000, 3000000000000, (300, True),
+                 (3000, True), (3000000, True), (300, False, True), (3000, False, True),
+                 (3000000, False, True), (1024, False, True), (10**26 * 30, False, True),
+                 (10**26 * 30, True), 10**26 * 30,
+                 (3141592, False, False, '%.2f'), (3000, False, True, '%.3f'),
+                 (3000000000, False, True, '%.0f'), (10**26 * 30, True, False, '%.3f'),)
+        results = ('300 Bytes', '3.0 kB', '3.0 MB', '3.0 GB', '3.0 TB',
+                   '300 Bytes', '2.9 KiB', '2.9 MiB', '300B', '2.9K', '2.9M', '1.0K', '2481.5Y',
+                   '2481.5 YiB', '3000.0 YB',
+                   '3.14 MB', '2.930K', '3G', '2481.542 YiB')
+        self.assert_many_results(bytes_util.naturalsize, tests, results)
```

Comment:
I would combine  the `tests` and `results`, i.e.

```
tests_and_results = [
  (300, '300 Bytes'),
  ((300, True), '300 Bytes')
]

for scenario in tests_and_results:
  test, expected_result = scenario
  result = byte_util.natural_size(test)
  self.assertEqual(expected_result, result)

```

But, it's up to you since this is style related feedback :-)